sycl/include/syclcompat/memory.hpp

```diff
@@ -85,6 +85,26 @@ enum memcpy_direction {
   device_to_device,
   automatic
 };
+
+inline __SYCL_ALWAYS_INLINE uint32_t nvvm_get_smem_pointer(void *ptr) {
```

Comment:
@Alcpz do you want me to include the defs.hpp file that defines the __syclcompat_inline__ macro in memory.hpp?

It's unclear to me what you want me to change. thanks

Reply:
Yes. It's fine to include `defs.hpp` here. Sorry If I send a confusing message due to my other comments. As long as you don't include all of `syclcompat.hpp`, it's fine to add the headers that provide the functionality you need.

Reply:
OK sure I'll do it like this. Isn't this quite confusing for users though? How are they meant to know that they can't use these new apis unless they also include defs.hpp?

Reply:
I mean shouldn't it be memory.hpp that includes defs.hpp, rather than relying on users having to include it in their code?

I think that there were similar pre-existing issues to this, which was the reason I needed to include the full syclcompat.hpp in the user test code I added in test-e2e.

Reply:
I've included defs.hpp in memory.hpp. Hope this is OK.